src/rpc.rs

```diff
@@ -31,12 +28,36 @@ pub enum Request {
 impl Request {
     /// Read a request from the given readable.
     pub fn read(conn: &mut impl Read, expected_chain_id: &chain::Id) -> Result<Self, Error> {
-        let msg_bytes = read_msg(conn)?;
-
-        // Parse Protobuf-encoded request message
-        let msg = proto::privval::Message::decode_length_delimited(msg_bytes.as_ref())
-            .map_err(|e| format_err!(ErrorKind::ProtocolError, "malformed message packet: {}", e))?
-            .sum;
+        let mut msg_bytes: Vec<u8> = vec![];
```

Comment:
not a rust expert but, if we know that in 99% of use cases (this is everything except sei) the buffer will be not bigger than max size, then maybe it is better to allocate the buffer with max size, [Vec::with_capacity](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity) ?

Reply:
Thanks! 

From my observations, everything except Sei fits into `DATA_MAX_SIZE`, -- this is why it doesn't fail, essentially. Therefore everything but Sei will read only one chunk. In this case, I don't see much difference, since it will be the same amount of allocations, right? Although if we move `msg_bytes` somewhere out of `read` so that we can reuse the buffer it might save us an allocation per read (same logic is applicable to the upstream tmkms as well, not specific to the aggregation case). 

Let me know if I'm missing something here.

Reply:
> I don't see much difference, since it will be the same amount of allocations, right
reading this: https://nnethercote.github.io/perf-book/heap-allocations.html?highlight=borrow#vec-growth

> A new, empty Vec created by the common means ([vec![]](https://doc.rust-lang.org/std/macro.vec.html) or [Vec::new](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.new) or [Vec::default](https://doc.rust-lang.org/std/default/trait.Default.html#tymethod.default)) has a length and capacity of zero, and no heap allocation is required. If you repeatedly push individual elements onto the end of the Vec, it will periodically reallocate. The growth strategy is not specified, but at the time of writing it uses a quasi-doubling strategy resulting in the following capacities: 0, 4, 8, 16, 32, 64, and so on

so yeah, `vec![]` is equivalent to zero allocation/no allocation. Therefore yeah, there will be no reallocations. Meaning that you can ignore my comment :)